lib/_http_client.js

```diff
@@ -87,6 +87,14 @@ function ClientRequest(options, cb) {
     }
     if (host && !this.getHeader('host') && setHost) {
       var hostHeader = host;
+
+      // For the Host header, ensure that IPv6 addresses are enclosed
+      // in square brackets, as defined by URI formatting
+      // https://tools.ietf.org/html/rfc3986#section-3.2.2
+      if (net.isIPv6(hostHeader)) {
+        hostHeader = '[' + hostHeader + ']';
```

Comment:
I want to say that some benchmarks have shown that template strings do not introduce any performance issues, maybe even improve performance. Will have to look and see if I can find them though.